src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useRef, useContext } from 'react';
+import './index.scss';
+
+import { LocalizationContext } from '../../../../lib/LocalizationContext';
+import Avatar from '../../../../ui/Avatar';
+import { ButtonTypes } from '../../../../ui/Button';
+import Modal from '../../../../ui/Modal';
+import Icon, { IconTypes, IconColors } from '../../../../ui/Icon';
+import Input, { InputLabel } from '../../../../ui/Input';
+import Label, { LabelColors, LabelTypography } from '../../../../ui/Label';
+import TextButton from '../../../../ui/TextButton';
+import { useCreateOpenChannelContext } from '../../context/CreateOpenChannelProvider';
+
+export interface CreateOpenChannelUIProps {
+  closeModal?: () => void;
+  renderHeader?: () => React.ReactElement;
+  renderProfileInput?: () => React.ReactElement;
+}
+
+function CreateOpenChannelUI({
+  closeModal,
+  renderHeader,
+  renderProfileInput,
+}: CreateOpenChannelUIProps): React.ReactElement {
+  const [newFile, setNewFile] = useState(null);
+  const [currentImage, setCurrentImg] = useState(null);
+  const inputFormRef = useRef(null);
+  const inputFileRef = useRef(null);
+  const inputTextRef = useRef(null);
+  const { stringSet } = useContext(LocalizationContext);
+  const {
+    logger,
+    createNewOpenChannel,
+  } = useCreateOpenChannelContext();
+  return (
+    <div className="sendbird-create-open-channel-ui">
+      <Modal
+        titleText={stringSet.CREATE_OPEN_CHANNEL_LIST__TITLE}
+        submitText={stringSet.CREATE_OPEN_CHANNEL_LIST__SUBMIT}
+        type={ButtonTypes.PRIMARY}
+        onCancel={closeModal}
+        renderHeader={renderHeader}
+        onSubmit={() => {
+          const channelName = inputTextRef?.current?.value;
```

Comment:
https://twitter.com/DavidKPiano/status/1569361453928300545?s=20&t=CJXyhVXgINPA5YVv2Rkqdw -> nit - do  want to try this method?

Reply:
yes, I will try in the next refactoring step